File: basic-java-app/app/src/main/java/com/driouechemed/basicjavaapp/database/AppDatabase.java
package com.driouechemed.basicjavaapp.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.driouechemed.basicjavaapp.database.dao.TaskDao;
import com.driouechemed.basicjavaapp.database.entities.Task;

@Database(entities = {Task.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase {

    private static final String DATABASE_NAME = "basic-app.db";
    private static AppDatabase instance;

    public static AppDatabase getAppDatabase(Context context) {
        if (instance == null) {
            Builder<AppDatabase> builder = Room.databaseBuilder(context.getApplicationContext(),
                    AppDatabase.class, DATABASE_NAME);
            instance = builder.build();
        }
        return instance;
    }

     public abstract TaskDao getTaskDao();
}

File: basic-kotlin-app/app/src/main/java/com/driouechemed/basickotlinapp/database/AppDatabase.kt
package com.driouechemed.basickotlinapp.database

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.driouechemed.basickotlinapp.database.dao.TaskDao
import com.driouechemed.basickotlinapp.database.entities.Task

@Database(entities = [Task::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun taskDao(): TaskDao
}

object DatabaseProvider {
    private const val DATABASE_NAME: String = "lean_kotlin"
    private var instance: AppDatabase? = null

    fun getInstance(context: Context): AppDatabase {
        if (instance == null) {
            instance =  Room.databaseBuilder(
                    context.applicationContext,
                    AppDatabase::class.java, DATABASE_NAME
                ).build()
        }
        return instance!!
    }
}
